x/profiles/keeper/keeper_chain_links.go

```diff
@@ -95,3 +94,64 @@ func (k Keeper) DeleteAllUserChainLinks(ctx sdk.Context, user string) {
 		k.DeleteChainLink(ctx, link)
 	}
 }
+
+// getOldestUserChainByChain returns the oldest chain link of the given owner associated to the given chain name
+// If the such the chain link is not exists, return false instead.
+func (k Keeper) getOldestUserChainByChain(ctx sdk.Context, owner, chainName string) (types.ChainLink, bool) {
+	var oldestLink types.ChainLink
+	found := false
+	k.IterateUserChainLinksByChain(ctx, owner, chainName, func(link types.ChainLink) (stop bool) {
+		if !found {
+			oldestLink = link
+			found = true
+			return false
+		}
+
+		if link.CreationTime.Before(oldestLink.CreationTime) {
+			oldestLink = link
+		}
+		return false
+	})
+
+	return oldestLink, found
+}
+
+// updateOwnerDefaultExternalAddress updates the default external address of the given chain name for the given owner
+// It must be performed after deleting the default external address chain link
+func (k Keeper) updateOwnerDefaultExternalAddress(ctx sdk.Context, owner, chainName string) {
+	store := ctx.KVStore(k.storeKey)
+	link, found := k.getOldestUserChainByChain(ctx, owner, chainName)
+	if found {
+		srcAddrData, err := types.UnpackAddressData(k.cdc, link.Address)
+		if err != nil {
+			panic(err)
+		}
+		k.SaveDefaultExternalAddress(ctx, owner, chainName, srcAddrData.GetValue())
+		return
+	}
+
+	// If owner has no chain link on the given chain name, then delete the key
+	store.Delete(types.DefaultExternalAddressKey(owner, chainName))
```

Comment:
I think it's better from a performance perspective putting the shorter branch inside an `if`: 

```suggestion
	if !found {
		// If the owner has no chain link on the given chain name, then delete the key
		store.Delete(types.DefaultExternalAddressKey(owner, chainName))
	}
	
	srcAddrData, err := types.UnpackAddressData(k.cdc, link.Address)
	if err != nil {
		panic(err)
	}
	k.SaveDefaultExternalAddress(ctx, owner, chainName, srcAddrData.GetValue())
```